route/test/test_route_generator.cpp

```diff
@@ -25,36 +25,83 @@
 #include <lanelet2_core/Attribute.h>
 #include <lanelet2_core/geometry/LineString.h>
 #include <lanelet2_core/primitives/Traits.h>
+#include <lanelet2_extension/traffic_rules/CarmaUSTrafficRules.h>
+#include <lanelet2_extension/utility/query.h>
+#include <lanelet2_extension/utility/utilities.h>
 #include <lanelet2_extension/projection/local_frame_projector.h>
+#include <lanelet2_extension/io/autoware_osm_parser.h>
+
 #include <gtest/gtest.h>
 #include <ros/ros.h>
 
+/*
+Using this file:
+    1) Libraries you should have: carma-base, carma-config, carma-msgs (integration/routing branch), 
+        carma-platform (integration/routing branch), carma-utils, caram-web-ui, opendrive2lanelet.
+    2) Then the test can be built with the command: 
+        ./carma_build -c /workspaces/carma_ws/carma/ -a /workspaces/carma_ws/autoware.ai/ -x -m "--only-pkg-with-deps route"
+    3) Update the osm file location and starting/ending IDs to match the file you want to test
+    4) Run the unit test with
+        catkin_make run_tests_route
+    5) Confirm that the test passed and that the list of lanelet IDs does traverse from the start to the end
+*/
+
 TEST(RouteGeneratorTest, testLaneletRouting)
 {
     tf2_ros::Buffer tf_buffer;
     carma_wm::WorldModelConstPtr wm;
     route::RouteGeneratorWorker worker(tf_buffer);
-    //lanelet::LaneletMapPtr map = lanelet::load("/home/qswawrq/Desktop/ATEF_pretty.osm",
-    //    lanelet::projection::LocalFrameProjector("+proj=tmerc +lat_0=39.46636844371259 +lon_0=-76.16919523566943 +k=1 +x_0=0 +y_0=0 +units=m +vunits=m"));
-    lanelet::LaneletMapPtr map = lanelet::load("/home/qswawrq/Desktop/ATEF_pretty.osm", lanelet::Origin({0, 0}));
+
+    int projector_type = 0;
+    std::string target_frame;
+    lanelet::ErrorMessages load_errors;
+
+    // If the output is an error about the geoReference field in the osm file, here is a correct example. If the lat/lon coordinates are already correct, simply add:
+    // <geoReference>+proj=tmerc +lat_0=0 +lon_0=0 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=m +geoidgrids=egm96_15.gtx +vunits=m +no_defs</geoReference>
+
+    // File location of osm file
+    std::string file = "/workspaces/carma_ws/carma/src/Town02_pretty.osm";
```

Comment:
Is this file going to exist in CI and on people's development machines? If it's an open-source test file (sounds like it's form CARLA so that should be the case) then we can add it to the test directory here if it's not super large. Should make the tests consistent.

Reply:
There is already a test_vector_map.osm file that I could change it to instead. My understanding is that the point of this file is for testing maps, so a user has their own map and changes the path and lanelet IDs to match, to confirm that that map is routable, rather than running it with an example file.

Reply:
I will change both links to be relative and to valid files just in case